pyccel/codegen/printing/cucode.py

```diff
@@ -61,6 +62,47 @@ def _print_Module(self, expr):
         self.exit_scope()
         return code
 
+    def function_signature(self, expr, print_arg_names = True):
+        """
+        Get the Cuda representation of the function signature.
+
+        Extract from the function definition `expr` all the
+        information (name, input, output) needed to create the
+        function signature and return a string describing the
+        function.
+        This is not a declaration as the signature does not end
+        with a semi-colon.
+
+        Parameters
+        ----------
+        expr : FunctionDef
+            The function definition for which a signature is needed.
+
+        print_arg_names : bool, default : True
+            Indicates whether argument names should be printed.
+
+        Returns
+        -------
+        str
+            Signature of the function.
+        """
+        cuda_decorater = '__global__' if('kernel' in expr.decorators) else ''
+        c_function_signature = super().function_signature(expr, print_arg_names)
+        return f'{cuda_decorater} {c_function_signature}'
+
+    def _print_KernelCall(self, expr):
+        func = expr.funcdef
+        args = []
+        for a in expr.args:
+            arg_val = a.value or Nil()
+            args.append(arg_val)
```

Comment:
List comprehensions tend to be faster than loops with append. Here The loops seems small enough that this should easily be able to be done easily:
```suggestion
        args = [a.value or Nil() for a in expr.args]
```

Reply:
[1a4527c](https://github.com/pyccel/pyccel-cuda/pull/42/commits/1a4527c27a20312339b6cc00138ad0512a4f9777)